src/Native/Unix/configure.cmake

```diff
@@ -33,24 +33,33 @@ check_include_files(
     HAVE_LINUX_IN_H)
 
 if (HAVE_LINUX_IN_H)
-    set (SOCKET_INCLUDES ${SOCKET_INCLUDES} linux/in.h)
+    set (SOCKET_INCLUDES linux/in.h)
 else ()
-    set (SOCKET_INCLUDES ${SOCKET_INCLUDES} netinet/in.h)
+    set (SOCKET_INCLUDES netinet/in.h)
 endif ()
 
-set(CMAKE_EXTRA_INCLUDE_FILES ${SOCKET_INCLUDES})
-
-check_type_size(
-    "struct in_pktinfo"
-    HAVE_IN_PKTINFO
-    BUILTIN_TYPES_ONLY)
+check_c_source_compiles(
+    "
+    #include <${SOCKET_INCLUDES}>
+    int main()
+    {
+        struct in_pktinfo;
+        return 0;
+    }
+    "
+    HAVE_IN_PKTINFO)
```

Comment:
If the HAVE_IN_PKTINFO test was not working, it would result in either missing or multiply defined in_pktinfo which would both lead to a build failure. We are defining the in_pktinfo ourselves if we don't detect that it is present. That was actually the issue on the older Linux - the previous way of testing didn't work and so the build was then failing due to multiply defined in_pktinfo here: https://github.com/dotnet/corefx/blob/7dfd5d8da4b7ec6e96988584fae491c438d7d1e7/src/Native/Unix/System.Native/pal_networking.cpp#L100.
I'll check the HAVE_IP_MREQN, this one could fail without causing build failure.